src/sqlfluff/core/rules/std/L029.py

```diff
@@ -29,21 +30,17 @@ class Rule_L029(BaseCrawler):
 
     """
 
-    config_keywords = ["only_aliases"]
+    config_keywords = ["unquoted_identifiers_policy"]
 
     def _eval(self, segment, dialect, parent_stack, **kwargs):
         """Keywords should not be used as identifiers."""
-        if segment.name == "naked_identifier":
-            # If self.only_aliases is true, we're a bit pickier here
-            if self.only_aliases:
-                # Aliases are ok (either directly, or in column definitions or in with statements)
-                if parent_stack[-1].is_type(
-                    "alias_expression", "column_definition", "with_compound_statement"
-                ):
-                    pass
-                # All other references may not be at the discretion of the developer, so leave them out
-                else:
-                    return None
-            # Actually lint
-            if segment.raw.upper() in dialect.sets("unreserved_keywords"):
-                return LintResult(anchor=segment)
+        if (
+            segment.name == "naked_identifier"
+            and unquoted_ids_policy_applicable(
+                self.unquoted_identifiers_policy, parent_stack
+            )
+            and (segment.raw.upper() in dialect.sets("unreserved_keywords"))
+        ):
+            return LintResult(anchor=segment)
+        else:
```

Comment:
nit: this else: return None isn't needed but if you prefer being explicit it's fine for me :)

Reply:
I tend to like explicit `return` statements. In my recent PRs, I've been adding MyPy type hints to several areas of the code, and IIRC, either PyCharm (my Python IDE) and/or MyPy complains if you don't explicitly return a value matching the function's return type.

Reply:
`return None` in sqlfluff seems to have a meaning that's essentially equivalent to `return SkipThis` so it's nice to be explicit about it for the reader's sake.